python/src/iceberg/table/partitioning.py

```diff
@@ -24,7 +24,7 @@ class PartitionField:
 
     Attributes:
         source_id(int): The source column id of table's schema
-        field_id(int): The partition field id across all the table metadata's partition specs
+        field_id(int): The partition field id across all the table partition specs' metadata
```

Comment:
This changes the meaning. "Table metadata" refers to all of the metadata, and "table metadata's partition specs" is the set of partition specs in table metadata. "Partition specs' metadata" is narrower, the metadata across partition specs.

To fix this, I think you can just omit "metadata": "across all the table's partition specs".

Reply:
Love it, less is more 👍🏻